fsspec/implementations/prefix.py

```diff
@@ -0,0 +1,135 @@
+import os
+from pathlib import Path
+from typing import Any, Iterable, Sequence, Union
+
+import fsspec
+from fsspec import AbstractFileSystem
+from fsspec.core import split_protocol
+from fsspec.utils import stringify_path
+
+
+class PrefixFileSystem(AbstractFileSystem):
+    def __init__(
+        self,
+        prefix: str,
+        filesystem: fsspec.AbstractFileSystem,
+        *args,
+        **storage_options,
+    ) -> None:
+        super().__init__(*args, **storage_options)
+        self.prefix = prefix
+        self.filesystem = filesystem
+
+    def _add_fs_prefix(self, path: Union[str, Path]) -> Union[str, Sequence[str]]:
+        if isinstance(path, (str, Path)):
```

Comment:
Type annotations say that `path` is str in all methods like `ls/makedirs/etc`, so we need to either remove it or propagate Union to those methods too (I suppose this is only relevant for LocalFileSystem right now, right?).

Reply:
I removed the `Union` annotation!

Reply:
But we still have the check here and I suppose this is all meant to work with localfs, which accepts Path-like objects too. Looks like there type annotations  are not yet ready (https://github.com/intake/filesystem_spec/issues?q=is%3Aissue+is%3Aopen+type+annotations), so maybe not worth adding them at all for now? 🤷 CC @martindurant

Reply:
@lucmos Your type annotation says `str`, but you check for `Path` too 🙁

Reply:
Ok I completely removed the `Path` and any `pathlib` import!